packages/design-system-react-native/src/primitives/ImageOrSvg/ImageOrSvg.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+import React, { useState, useCallback, useEffect } from 'react';
+import { Image } from 'react-native';
+import { SvgUri } from 'react-native-svg';
+
+import type { ImageOrSvgProps } from './ImageOrSvg.types';
+
+const ImageOrSvg = ({
+  src,
+  width,
+  height,
+  onImageLoad,
+  onImageError,
+  onSvgError,
+  style,
+  imageProps,
+  svgProps,
+}: ImageOrSvgProps) => {
+  // CASE 1: local image (src is a number)
+  if (typeof src === 'number') {
+    return (
+      <Image
+        source={src}
+        style={[{ width, height } as any, style]}
+        resizeMode="contain"
+        onLoad={onImageLoad}
+        onError={onImageError}
+        {...imageProps}
+      />
+    );
+  }
+
+  // CASE 2: Local SVG component (src is a React component)
+  if (typeof src === 'function') {
+    const LocalSvg = src;
+    return <LocalSvg width={width} height={height} {...svgProps} />;
+  }
+
+  // CASE 3: Remote image or SVG (src is an object with a uri)
+  const [isSvg, setIsSvg] = useState<boolean>(false);
+
+  const checkSvgContentType = useCallback(async (uri: string) => {
+    try {
+      const response = await fetch(uri, { method: 'HEAD' });
+      // If no header is returned, fallback to an empty string.
+      const contentType = response.headers.get('Content-Type') || '';
+      return contentType.includes('image/svg+xml');
+    } catch {
+      return false;
+    }
+  }, []);
```

Comment:
non-blocking: I'm not very well experienced with caching but there may be some performance benefits to caching the HEAD request results outside the component. It could help eliminate redundant requests when the same SVG URI is used.

Reply:
This was carried over from the logic in [AvatarFavicon](https://github.com/MetaMask/metamask-mobile/blob/main/app/component-library/components/Avatars/Avatar/variants/AvatarFavicon/AvatarFavicon.tsx) from mobile, but it's a valid question. For now I'll merge this to unblock, if Cal comes back with an answer that we need to cache then I'll create a follow-up ticket/PR.

@Cal-L do you think we need to add caching logic here?